python_modules/dagster/dagster/core/asset_defs/decorators.py

```diff
@@ -322,6 +346,13 @@ def inner(fn: Callable[..., Any]) -> AssetsDefinition:
             asset_keys_by_output_name=asset_keys_by_output_name,
             node_def=op,
             asset_deps={asset_keys_by_output_name[name]: asset_deps[name] for name in asset_deps},
+            partitions_def=partitions_def,
+            partition_mappings={
+                asset_keys_by_input_name[input_name]: partition_mapping
```

Comment:
The tests are complaining that `asset_keys_by_input_name` isn't defined before it's referenced here.